dapp-example/src/App.js

```diff
@@ -0,0 +1,536 @@
+  const [contractCallViewMsg, setContractCallViewMsg] = useState(null);
+  const [contractCallUpdateMsg, setContractCallUpdateMsg] = useState(null);
+  const [htsTokenAddress, setHtsTokenAddress] = useState('');
+  const [htsTokenReceiverAddress, setHtsTokenReceiverAddress] = useState('');
+  const [htsTokenAmount, setHtsTokenAmount] = useState();
+  const [htsTokenMsg, setHtsTokenMsg] = useState(null);
+
+  const { recoveredPublicKeyToAccountId, getAccountInfo } = useHederaSdkClient();
+
+  useEffect(() => {
+    if (window.ethereum) {
+      const provider = new ethers.providers.Web3Provider(window.ethereum, 'any');
+
+      setSigner(provider.getSigner());
+
+      window.ethereum.on("accountsChanged", changeConnectedAccount);
+      window.ethereum.on("chainChanged", () => {
+        setErrorMessage(null);
+        setAddress(null);
+        setBalance(null);
+        setAccountId(null);
+        setContractAddress(null);
+        setIsLoading(null);
+        setDeployContractMsg(null);
+        setContractCallViewMsg(null);
+        setContractCallUpdateMsg(null);
+        setHtsTokenAddress(null);
+        setHtsTokenReceiverAddress(null);
+        setHtsTokenAmount(null);
+        setHtsTokenMsg(null);
+      });
+    }
+  }, []);
+
+  const isConnected = useMemo(() => {
+    return !!signer && !!address;
+  }, [signer, address]);
+
+  const isAccountActivated = useMemo(() => {
+    return address && Number(balance) > 0;
+  }, [address, balance]);
+
+  const fetchAccountBalance = async (address) => {
+    try {
+      // 
+      const balance = await window.ethereum.request({
+        method: "eth_getBalance",
+        params: [address.toString(), "latest"],
+      });
+      setBalance(ethers.utils.formatEther(balance));
+
+      // get account balance from SDK()
+      // const info = await getAccountInfo(address);
+
+      // setBalance(info.balance.toBigNumber().toNumber())
+    } catch (error) {
+      console.log(error);
+    }
+  };
+
+  const changeConnectedAccount = async (newAddress) => {
+    try {
+      newAddress = Array.isArray(newAddress) ? newAddress[0] : newAddress;
+
+      await fetchAccountBalance(newAddress);
+
+      setAddress(newAddress);
+      setAccountId(null);
+    } catch (err) {
+      console.error(err);
+      setErrorMessage("There was a problem connecting to MetaMask");
+    }
+  };
+
+  const connectAccountHandler = useCallback(async () => {
+    if (window.ethereum) {
+      try {
+        const accounts = await window.ethereum.request({
+          method: "eth_requestAccounts",
+        });
+
+        await changeConnectedAccount(accounts[0]);
+      } catch (err) {
+        console.error(err);
+        setErrorMessage("There was a problem connecting to MetaMask");
+      }
+    } else {
+      setErrorMessage("Install MetaMask");
+    }
+  }, []);
+
+  const showAccountIdHandler = useCallback(async () => {
+    const message = 'Welcome to Hedera';
```

Comment:
Can we update this message so it's dynamic? Possibly Include the evm account ID and a date/time stamp?